src/Workspaces/Core/Portable/Workspace/Solution/Project.cs

```diff
@@ -202,23 +205,23 @@ public bool ContainsAnalyzerConfigDocument(DocumentId documentId)
         /// <summary>
         /// Get the documentId in this project with the specified syntax tree.
         /// </summary>
-        public DocumentId GetDocumentId(SyntaxTree syntaxTree)
+        public DocumentId? GetDocumentId(SyntaxTree? syntaxTree)
```

Comment:
💭 Seems strange to allow null input here, especially considering the method with `GetDocument(DocumentId)` does not allow a null input.

Reply:
Agreed, I don't like that this allows a null, but alas that's the contract being implemented. I'm all for considering _changing_ the contract, but that's orthogonal from documenting the contract we have today.